src/Neo.VM/ReferenceCounterV2.cs

```diff
@@ -0,0 +1,105 @@
+// Copyright (C) 2015-2024 The Neo Project.
+//
+// ReferenceCounterV2.cs file belongs to the neo project and is free
+// software distributed under the MIT software license, see the
+// accompanying file LICENSE in the main directory of the
+// repository or http://www.opensource.org/licenses/mit-license.php
+// for more details.
+//
+// Redistribution and use in source and binary forms with or without
+// modifications are permitted.
+
+using Neo.VM.StronglyConnectedComponents;
+using Neo.VM.Types;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using Array = Neo.VM.Types.Array;
+using Buffer = Neo.VM.Types.Buffer;
+
+namespace Neo.VM
+{
+    /// <summary>
+    /// Used for reference counting of objects in the VM.
+    /// </summary>
+    public sealed class ReferenceCounterV2 : IReferenceCounter
+    {
+        // Keeps the total count of references.
+        private int _referencesCount = 0;
+
+        /// <inheritdoc/>
+        public int Count => _referencesCount;
+
+        /// <inheritdoc/>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static bool NeedTrack(StackItem item)
+        {
+            throw new NotImplementedException();
```

Comment:
If you do nothing instead of throwing exception, no changes will required in any type and it will be compatible with both systems

Reply:
need track is private, so should be fine, it will not be used anyway.